### DISCUSSION TITLE
Failed to send message. Error: Pass options.removeUndefinedValues=true to remove undefined values from map/array/set.

### DISCUSSION THREAD

**Discussion by lukman-bahar-ap** [kirimwa-id/forum#95] [Tanya Jawab] [answered]
Hi mas Rio,
Saya tidak bisa send messages dengan error berikut, menggunakan php 5 curl seperti biasa, masih bingung penyebabnya apa? 

> Failed to send message. Error: Pass options.removeUndefinedValues=true to remove undefined values from map/array/set.

Kode php 5.6 curl : 

```
>  $url = "https://api.kirimwa.id/v1/messages";
>  $paramspayload = array(
>    "device_id" => "asus-zenfone-10",
>    "message" => "test \n test wa",
>    "phone_number" => "[nomor dirahasiakan]",
>    "message_type" => "text");
>  
>  $authorization = "Authorization: Bearer  [token dirahasiakan]";
> 
> $ch = curl_init($url);
>  curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, 0);
>  curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 0);
>  curl_setopt($ch, CURLOPT_POST, 1);
>  curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($paramspayload)); 
>  curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json' , $authorization));
>  curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
>  curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 0);
>  $result = curl_exec($ch);
>  curl_close($ch);
>  print_r ($result);
>  echo json_encode($paramspayload);

```
output : 

> {"message":"Failed to send message. Error: Pass options.removeUndefinedValues=true to remove undefined values from map/array/set."}{"device_id":"mtsn2jember-hp-1","message":"test \n test wa","phone_number":"[EDITED]","message_type":"text"}

**Comment by rioastamal** [comment, marked as answer]
@lukman-bahar-ap , Error tersebut memang sepertinya dari server. Saya tidak memiliki PHP 5.6 tapi kode diatas saya coba menggunakan PHP 7.4 berhasil di komputer saya. Apakah error tersebut bersifat acak atau selalu terjadi?

Coba lagi menggunakan versi 7.4 jika masih terjadi coba gunakan fungsi `apiKirimWaRequest()` yang ada di dokumentasi [referensi API](https://developer.kirimwa.id/api/v1/?php#contoh-integrasi).
